tests_e2e/tests/scripts/agent_publish-check_update.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env pypy3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import re
+import sys
+import logging
+
+from tests_e2e.tests.lib.agent_log import AgentLog
+from tests_e2e.tests.lib.retry import retry_if_false
+
+
+# pylint: disable=W0105
+"""
+Post the _LOG_PATTERN_00 changes, the last group sometimes might not have the 'Agent' part at the start of the sentence; thus making it optional.
+
+> WALinuxAgent-2.2.18 discovered WALinuxAgent-2.2.47 as an update and will exit
+(None, 'WALinuxAgent-2.2.18', '2.2.47')
+"""
+_UPDATE_PATTERN_00 = re.compile(r'(.*Agent\s)?(\S*)\sdiscovered\sWALinuxAgent-(\S*)\sas an update and will exit')
+
+"""
+> Agent WALinuxAgent-2.2.45 discovered update WALinuxAgent-2.2.47 -- exiting
+('Agent', 'WALinuxAgent-2.2.45', '2.2.47')
+"""
+_UPDATE_PATTERN_01 = re.compile(r'(.*Agent)?\s(\S*) discovered update WALinuxAgent-(\S*) -- exiting')
+
+"""
+> Normal Agent upgrade discovered, updating to WALinuxAgent-2.9.1.0 -- exiting
+('Normal Agent', None, 'WALinuxAgent-2.9.1.0 ')
+"""
+_UPDATE_PATTERN_02 = re.compile(r'(.*Agent)?\s(\S*) upgrade discovered, updating to WALinuxAgent-(\S*) -- exiting')
```

Comment:
```suggestion
_UPDATE_PATTERN_02 = re.compile(r'(.*Agent) upgrade discovered, updating to (WALinuxAgent)-(\S*) -- exiting')
```

Reply:
I have a PR open on the DCR repo to fix a bug with this reg ex, it wasn't reviewed or merged but we can make the change directly here -> https://msazure.visualstudio.com/One/_git/Compute-Runtime-Tux-Pipeline/pullrequest/8215795?_a=files